### PR TITLE
Add support for PostgreSQL 15 and 16

### PR DESCRIPTION
**Pull request by jonesbusy**
I'm not sure if we have some automated tests, but I can only confirm it works

pg_15_warn

### Testing done

See

PG_15

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by bguerin** [comment]
FYI integration test is done here:
https://github.com/jenkinsci/pipeline-maven-plugin/blob/master/pipeline-maven-database/src/test/java/org/jenkinsci/plugins/pipeline/maven/db/PipelineMavenPluginPostgreSqlDaoIT.java#L44

use latest postgres docker image, so actually, version 16 :)

File: pipeline-maven-database/src/main/java/org/jenkinsci/plugins/pipeline/maven/db/AbstractPipelineMavenPluginDao.java [hunk 1]
```diff
@@ -344,6 +344,7 @@ public FormValidation validateConfiguration(Config config) {
                                 }
                             }
                             switch (metaData.getDatabaseMajorVersion()) {
+                                case 15:
```
Comment (bguerin) [inline pipeline-maven-database/src/main/java/org/jenkinsci/plugins/pipeline/maven/db/AbstractPipelineMavenPluginDao.java:347]: please also add 16, as it is released and already tested

Comment (jonesbusy) [reply pipeline-maven-database/src/main/java/org/jenkinsci/plugins/pipeline/maven/db/AbstractPipelineMavenPluginDao.java:347]: Sure!

**Review by bguerin — CHANGES_REQUESTED** [review]
Please fix documentation also:
https://github.com/jenkinsci/pipeline-maven-plugin/blob/master/README.adoc?plain=1#L564
" Tested with PostgreSQL up to 16"

and the configuration help:
https://github.com/jenkinsci/pipeline-maven-plugin/blob/master/pipeline-maven/src/main/resources/org/jenkinsci/plugins/pipeline/maven/GlobalPipelineMavenConfig/help-jdbcUrl.html#L3C119-L3C129

**Review by bguerin — APPROVED** [review]

**Comment by bguerin** [comment]
Many thanks ! One thing out of my todo list :)

Auto merge enabled, will be released then

**Comment by bguerin** [comment]
@jonesbusy could you please re force push your branch please ? without any new change
Jenkins miss this PR because it was during the update time window, so it was down

and without a green build, no automerge and no release :sob:

**Comment by jonesbusy** [comment]
@bguerin Looks build is now ok but proably auto-merge was cancelled due to new push

**Pull request merged**